### PR TITLE
InputCommon: Eliminate some duplicated button threshold logic.

### PR DESCRIPTION
**Pull request by jordan-woyak**
Added `GetState` function to `Control` towards allowing `control_ref` member to be private.

### CODE REVIEW

File: Source/Core/InputCommon/ControlReference/ControlReference.h [hunk 1]
```diff
@@ -52,11 +52,17 @@ class ControlReference
 template <>
 inline bool ControlReference::GetState<bool>()
 {
-  return State() > ciface::ExpressionParser::CONDITION_THRESHOLD;
+  return std::lround(State()) > 0;
```
Comment (Tilka) [inline Source/Core/InputCommon/ControlReference/ControlReference.h]: I'd be careful with std::lround(), GCC is the only compiler that optimizes this to a movsx. Clang and MSVC both emit a float conversion and a call into the stdlib.

Comment (Tilka) [reply Source/Core/InputCommon/ControlReference/ControlReference.h]: But now I'm wondering why you convert to long.

Comment (jordan-woyak) [reply Source/Core/InputCommon/ControlReference/ControlReference.h]: @Tilka I'm just using `std::lround` to round to the nearest of 0 or 1.
That seemed cleaner to me than a 0.5 constant.

Comment (Tilka) [reply Source/Core/InputCommon/ControlReference/ControlReference.h]: Checking the value directly seems clearer to me, but I'm not the one who maintains this code^^

Comment (jordan-woyak) [reply Source/Core/InputCommon/ControlReference/ControlReference.h]: I'll add a comment so it's clear what's happening.

**Pull request merged**